aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -652,6 +652,81 @@ static void leaky_qrelu_out_kernel(Tensor& out, const Tensor& qx,
   });
 }
 
+static void qprelu_out_kernel(Tensor& out,
+                              const Tensor& qx,
+                              const Tensor& qw) {
+  int32_t i_zp = static_cast<int32_t>(qx.q_zero_point());
+  float i_scale = static_cast<float>(qx.q_scale());
+
+  int32_t w_zp = static_cast<int32_t>(qw.q_zero_point());
+  float w_scale = static_cast<float>(qw.q_scale());
+
+  int32_t o_zp = static_cast<int32_t>(out.q_zero_point());
+  float o_scale = static_cast<float>(out.q_scale());
+  float o_inv_scale = 1.0f / o_scale;
+
+  float multiplier = i_scale * w_scale * o_inv_scale;
+
+  int64_t input_ndim = qx.dim();
+  TORCH_CHECK(input_ndim > 0, "qprelu: zero-dim input tensor is not allowed.");
+
+  // Helper to convert 1d tensors or scalar tensor to an nd tensor that broadcasts with input
+  // All elements go into the channel dimension
+  DimVector sizes(input_ndim, 1), strides(input_ndim, 0);
+  auto as_nd = [&](const Tensor& t) {
+    TORCH_INTERNAL_ASSERT(t.defined() && (t.dim() == 1 || t.dim() == 0));
+    sizes[1] = t.dim() == 1 ? t.sizes()[0] : 1;
```

Comment:
cc @Xia-Weiwen wondering if you have tested this locally or not, looks like this can be index out of range error when the input is one dimensional

Reply:
Hi @jerryzh168 @z-a-f Thanks for pointing out this flaw and sorry for the negligence.
The following UT has covered the case where `input_ndim=1`: https://github.com/pytorch/pytorch/blob/b472852778eef777ac182e2efcc008429f5fc4fe/test/quantization/core/test_quantized_op.py#L474
Unfortunately, it fails to discover this flaw. I guess it is due to the implementation of `DimVector`.
I have created a PR to fix this: https://github.com/pytorch/pytorch/pull/82393. Please review. Thanks!